agent/engine/docker_container_engine.go

```diff
@@ -39,7 +39,8 @@ import (
 )
 
 const (
-	dockerDefaultTag = "latest"
+	dockerDefaultTag          = "latest"
+	containerEventsBufferSize = 1000
```

Comment:
Why `1000` ? 
Could you please add more details here ?

Reply:
1000 is picked up due to the maximum number of containers can be ran on an instance is ~1000. Let me know if you have better ideas.

Reply:
That's not accurate.  There's a limit of 1023 containers using the `bridge` network mode, but there is no limit of containers using `host` or `none`.

Reply:
A `note/comment` is here going to be useful. 
It would nice to calculate this value rather than using a fixed constant imho.